director/director_api.go

```diff
@@ -33,3 +49,78 @@ func ListNamespacesFromOrigins() []NamespaceAd {
 	}
 	return namespaces
 }
+
+func LoadDirectorPublicKey() (*jwk.Key, error) {
+	directorUrlStr := param.Federation_DirectorUrl.GetString()
+	if len(directorUrlStr) == 0 {
+		return nil, errors.Errorf("Director URL is unset; Can't load director's public key")
+	}
+	log.Debugln("Director URL:", directorUrlStr)
+	directorUrl, err := url.Parse(directorUrlStr)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintln("Invalid director URL:", directorUrlStr))
+	}
+	directorUrl.Scheme = "https"
+
+	discoveryUrl, _ := url.Parse(directorUrl.String())
+	discoveryUrl.Path = path.Join(".well-known/pelican-configuration", discoveryUrl.Path)
+
+	tr := config.GetTransport()
+	client := &http.Client{Transport: tr, Timeout: time.Second * 5}
+
+	req, err := http.NewRequest(http.MethodGet, discoveryUrl.String(), nil)
+	if err != nil {
+
+		return nil, errors.Wrap(err, fmt.Sprintln("Failure when doing director metadata request creation for: ", discoveryUrl))
```

Comment:
I don't think we actually need the sprintln:
`return nil, errors.Wrapf(err, "Failure when doing director metadata request creation for: %s", discoveryUrl.String()`